pkg/storage/stores/shipper/storage/client.go

```diff
@@ -0,0 +1,94 @@
+package storage
+
+import (
+	"context"
+	"io"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/grafana/loki/pkg/storage/chunk"
+)
+
+const delimiter = "/"
+
+type Client interface {
+	ListTables(ctx context.Context) ([]string, error)
+	ListFiles(ctx context.Context, tableName string) ([]IndexFile, error)
+	GetFile(ctx context.Context, tableName, fileName string) (io.ReadCloser, error)
+	PutFile(ctx context.Context, tableName, fileName string, file io.ReadSeeker) error
+	DeleteFile(ctx context.Context, tableName, fileName string) error
+	IsObjectNotFoundErr(err error) bool
```

Comment:
I am wondering if this should be `IsObjectNotFound` the signature makes it obvious it takes an error type.

When I first saw this I was confused what exactly this method was doing, I think removing the `Err` from the name makes it more clear to me